Add support for dynamic logstash plugins

dpb587 opened the pull request:
So, nearly the same as we handle elasticsearch plugins:

```
properties:
  # values are arrays, first being the installed plugin name, second being install source
  logstash_parser.plugins:
    # one of the built-in, official plugins
    - { "logstash-filter-punct" : "logstash-filter-punct" }
    # an internal plugin
    - { "cityindex-logstash-filter-acctlookup" : "https://s3.amazonaws.com/.../cityindex-logstash-filter-acctlookup-1.2.9.gem" }
```

dpb587 committed:
Add support for dynamic logstash plugins

dpb587 commented:
This was deployed to meta+cityindex.

Review, merge, and create release as ready, @mrdavidlaing.

mrdavidlaing commented on jobs/log_parser/spec:
```diff
@@ -52,6 +52,9 @@ properties:
     description: "Run a local elasticsearch client node"
     default: true
+  logstash_parser.plugins:
+    description: "Plugins to run logstash with (array[] = { plugin-name: install-source }"
```
@dpb587 - this description `(array[] = { plugin-name: install-source }` confuses me.  Is it possible to give a more verbose description?, eg:

```
  # values are arrays, first being the installed plugin name, second being install source
  logstash_parser.plugins:
    # one of the built-in, official plugins
    - [ "logstash-filter-punct", "logstash-filter-punct" ]
    # an internal plugin
    - [ "cityindex-logstash-filter-acctlookup", "https://s3.amazonaws.com/.../cityindex-logstash-filter-acctlookup-1.2.9.gem" ]
```

dpb587 committed:
Fix logstash plugin property to be key-value based

dpb587 commented:
@mrdavidlaing, I added an example to the property description (and to elasticsearch where I originally copied it from). In general, I don't feel like the spec property description is a very good place for documentation so I was trying to keep it concise. It also made me realize I implemented the logstash plugins slightly differently, so I switched to key-value (like elasticsearch) so they're the same.

mrdavidlaing commented:
That's much clearer to me.

One advantage to the `spec.description:` is that it gets exposed on bosh.io

![screen shot 2015-05-06 at 19 15 40](https://cloud.githubusercontent.com/assets/227505/7500211/4f804a46-f424-11e4-8690-cc0e3d31d013.png)

mrdavidlaing commented:
anywho; I'm happy for you to merge when ready
